routers/common/actions.go

```diff
@@ -5,77 +5,54 @@ package common
 
 import (
 	"fmt"
-	"net/http"
 	"strings"
 
 	actions_model "code.gitea.io/gitea/models/actions"
 	repo_model "code.gitea.io/gitea/models/repo"
 	"code.gitea.io/gitea/modules/actions"
+	"code.gitea.io/gitea/modules/util"
 	"code.gitea.io/gitea/services/context"
 )
 
-func DownloadActionsRunJobLogsWithIndex(ctx *context.Base, ctxRepo *repo_model.Repository, runID, jobIndex int64) {
-	if runID == 0 {
-		ctx.HTTPError(http.StatusBadRequest, "invalid run id")
-		return
-	}
-
+func DownloadActionsRunJobLogsWithIndex(ctx *context.Base, ctxRepo *repo_model.Repository, runID, jobIndex int64) error {
 	runJobs, err := actions_model.GetRunJobsByRunID(ctx, runID)
 	if err != nil {
-		ctx.HTTPError(http.StatusInternalServerError, err.Error())
-		return
+		return fmt.Errorf("GetRunJobsByRunID: %w", err)
 	}
-	if len(runJobs) == 0 {
-		ctx.HTTPError(http.StatusNotFound)
-		return
+	if err = runJobs.LoadRepos(ctx); err != nil {
+		return fmt.Errorf("LoadRepos: %w", err)
 	}
-	if err := runJobs.LoadRepos(ctx); err != nil {
-		ctx.HTTPError(http.StatusInternalServerError, err.Error())
-		return
-	}
-
-	var curJob *actions_model.ActionRunJob
-	if jobIndex >= 0 && jobIndex < int64(len(runJobs)) {
-		curJob = runJobs[jobIndex]
+	if 0 < jobIndex || jobIndex >= int64(len(runJobs)) {
```

Comment:
Caused #34038, It should be `jobIndex < 0`.  Fixed in #34041

Reply:
Sorry I made this typo. But why tests didn't catch it?